src/isotp_fast.c

```diff
@@ -1027,23 +1108,35 @@ int isotp_fast_send(struct isotp_fast_ctx *ctx, const uint8_t *data, size_t len,
             return ISOTP_N_BUFFER_OVERFLW;
         }
         struct isotp_fast_send_ctx *context;
-        int ret = get_send_ctx(ctx, rx_can_id, &context);
+        int ret = get_send_ctx(ctx, target_addr, &context);
         if (ret) {
             return ISOTP_NO_NET_BUF_LEFT;
         }
-        context->ctx = ctx;
-        context->tx_can_id = rx_can_id;
         context->data = data;
-        context->bs = ctx->opts->bs;
-        context->stmin = ctx->opts->stmin;
         context->rem_len = len;
-        context->state = ISOTP_TX_SEND_FF;
         context->cb_arg = cb_arg;
-        k_sem_init(&context->sem, 0, 1);
-        k_work_init(&context->work, send_work_handler);
-        k_timer_init(&context->timer, send_timeout_handler, NULL);
 
         k_work_submit(&context->work);
     }
     return ISOTP_N_OK;
 }
+
+#ifdef CONFIG_ISOTP_FAST_FIXED_ADDRESSING
+int isotp_fast_send_fixed(struct isotp_fast_ctx *ctx, const uint8_t *data, size_t len,
+                          const uint8_t target_addr, void *cb_arg)
+{
+    struct isotp_fast_addr rx_can_id = ctx->rx_can_id;
+    rx_can_id.ext_id &= target_addr;
+    struct isotp_fast_addr tx_can_id = isotp_fast_get_counterpart_addr(rx_can_id);
+    return isotp_fast_send(ctx, data, len, tx_can_id, cb_arg);
+}
+
+struct isotp_fast_addr isotp_fast_get_counterpart_addr(const struct isotp_fast_addr addr)
+{
+    return (struct isotp_fast_addr){
+        .ext_id = (addr.ext_id & ISOTP_FIXED_ADDR_PRIO_MASK) | (addr.ext_id & 0xFF0000)
+                  | ((addr.ext_id & ISOTP_FIXED_ADDR_TA_MASK) >> ISOTP_FIXED_ADDR_TA_POS)
+                  | ((addr.ext_id & ISOTP_FIXED_ADDR_SA_MASK) << ISOTP_FIXED_ADDR_TA_POS)
```

Comment:
It's only a coincidence that this works. The correct swapping operation using the defines should look like this:

```suggestion
                  | (((addr.ext_id & ISOTP_FIXED_ADDR_TA_MASK) >> ISOTP_FIXED_ADDR_TA_POS) << ISOTP_FIXED_ADDR_SA_POS)
                  | (((addr.ext_id & ISOTP_FIXED_ADDR_SA_MASK) >> ISOTP_FIXED_ADDR_SA_POS) << ISOTP_FIXED_ADDR_TA_POS)
```

Or use temporary variables to make more clear what's done here.